data_safe_haven/external/api/graph_api.py

```diff
@@ -770,18 +769,17 @@ def http_delete(self, url: str, **kwargs: Any) -> requests.Response:
                 timeout=120,
                 **kwargs,
             )
-            # We do not use response.ok as this allows 3xx codes
-            if (
-                requests.codes.OK
-                <= response.status_code
-                < requests.codes.MULTIPLE_CHOICES
-            ):
-                return response
-            raise DataSafeHavenInternalError(response.content)
-        except Exception as exc:
+        except requests.exceptions.RequestException as exc:
             msg = f"Could not execute DELETE request to '{url}'."
             raise DataSafeHavenMicrosoftGraphError(msg) from exc
 
+        # We do not use response.ok as this allows 3xx codes
+        if requests.codes.OK <= response.status_code < requests.codes.MULTIPLE_CHOICES:
+            return response
+        else:
+            msg = f"Could not execute DELETE request to '{url}'. Response content received: '{response.content.decode()}'."
```

Comment:
```suggestion
            msg = f"Could not execute DELETE request to '{url}'. Response content received: '{response.text}'."
```

I think this is because `response.content` is a `bytes` object, but the exception expects a `str`. I think `Response.text` is a `str`.
If this works it could be applied across all the places.